### PR TITLE
[`Optimization`] Parsing Smart Contract Script Analysis

### PR DESCRIPTION
**Pull request by cschuchardt88**
# Change Log
 - Added `VMInstruction` class (_faster than `Instruction` class_)
 - Changed `Parse` command output format and colors.

image

## Type of change

- [x] Optimization (the change is only an optimization)
- [x] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Local testing in terminal

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Jim8y** [comment]
a little bit explanation of how you achieved it would be great.

**Comment by cschuchardt88** [comment]
> a little bit explanation of how you achieved it would be great.

I don't follow. Code speak for itself?

**Comment by Jim8y** [comment]
> > a little bit explanation of how you achieved it would be great.
> 
> I don't follow. Code speak for itself?

Its up to you. But i personally hate `Code speak for itself?`, it has made the project hard to maintain and hard to use already. Not saying yours is bad.

**Comment by Jim8y** [comment]
Sort of like, yes, we can have it, but not really necessary one.

**Review by Jim8y — APPROVED** [review]

**Review by shargon — COMMENTED** [review]
How much it optimizes? having two classes to parse an instruction seems like a usability cost to me, so I would like to know how much it optimizes, I think it would be better to optimize the Instruction class, since it is the one that is really usually used

**Comment by cschuchardt88** [comment]
> How much it optimizes? having two classes to parse an instruction seems like a usability cost to me, so I would like to know how much it optimizes, I think it would be better to optimize the Instruction class, since it is the one that is really usually used

Yes, I want to do change `Instruction` class to the one I made here, but isnt complete yet for all features of the `Instruction` class. But I guess I could complete it. instead of this pr.

**Comment by cschuchardt88** [comment]
Add more `OpCode` formatting

image
image

**Review by Jim8y — DISMISSED** [review]

File: src/Neo.CLI/CLI/MainService.Tools.cs [hunk 1]
```diff
@@ -417,62 +418,81 @@ private static string Base64Fixed(string str)
...
-                    {
-                        result.Add($"{op}");
-                    }
-                }
-            }
-            return Environment.NewLine + string.Join("\r\n", result.ToArray());
+            var operand = instruction.Operand[instruction.OperandPrefixSize..].ToArray();
+            var asStr = Encoding.UTF8.GetString(operand);
+            var readable = asStr.All(char.IsAsciiLetterOrDigit);
+            return instruction.OpCode switch
```
Comment (shargon) [inline src/Neo.CLI/CLI/MainService.Tools.cs:449]: Move to .ToString()?

Comment (cschuchardt88) [reply src/Neo.CLI/CLI/MainService.Tools.cs:449]: It does return a `string`. However I am checking it before that.

**Review by vncoelho — COMMENTED** [review]
Looks good to me, I will soon test

File: src/Neo.CLI/CLI/MainService.Tools.cs (hunk 1 above)
Comment (shargon) [reply src/Neo.CLI/CLI/MainService.Tools.cs]: Moved

**Comment by shargon** [comment]
@cschuchardt88 please take a look to my last changes

**Review by shargon — APPROVED** [review]

**Comment by cschuchardt88** [comment]
looks good

**Review by Jim8y — APPROVED** [review]

**Pull request merged**